opensquirrel/circuit_builder.py

```diff
@@ -128,11 +129,18 @@ def _check_generator_f_args(
                 msg = "unknown annotation type"
                 raise TypeError(msg) from e
 
-            if not isinstance(args[i], expected_type):
+            # fix for python39
+            try:
+                is_incorrect_type = not isinstance(args[i], expected_type)
+            except TypeError:
+                # expected type is probably a Union, which works differently in python39
+                is_incorrect_type = not isinstance(args[i], expected_type.__args__)
+
+            if is_incorrect_type:
                 msg = f"wrong argument type for instruction `{attr}`, got {type(args[i])} but expected {expected_type}"
                 raise TypeError(msg)
-            if args[i].__class__.__name__ == "Qubit":
-                self._check_qubit_out_of_bounds_access(args[i].index)
+            if expected_type in (QubitLike, Qubit):
```

Comment:
Can this be simplified to `expected_type in QubitLike`?

Reply:
This does not work because `TypeError: argument of type '_UnionGenericAlias' is not iterable` 

Placing .__args__ introduces regex issues in the writer.

Reply:
Cool. Thanks for checking!